server/build.gradle

```diff
@@ -110,6 +109,9 @@ dependencies {
   annotationProcessor "org.apache.logging.log4j:log4j-core:${versions.log4j}"
   annotationProcessor project(':libs:opensearch-common')
 
+  // gson
+  api 'com.google.code.gson:gson:2.10.1'
```

Comment:
@abseth-amzn we use Jackson already, why do we need to bring GSON as well?

Reply:
The server module doesn't declare a dependency on jackson or gson. I saw gson being used in a bunch of other modules, hence used it. Hoping thats ok.

Reply:
It does not declare the dependency explicitly but it is there, coming from `opensearch-x-content`, `openseach-core`, the good part - it is coming in `lib/` folder already so we are not risking to break anyone with this new dependency (by JARhelling), thank you.

Reply:
Thanks @reta. Those modules are bringing in jackson-core and jackson-dataformat, but not jackson-databind which would be needed here.
Also, jackson does not allow unquoted field values, so following specifications wont be allowed as quotes would be required around the version and operator:
```
dependencies={ opensearch: x.y.z }
dependencies={ opensearch: =x.y.z }
dependencies={ opensearch: ~x.y.z }
```
gson allows for some flexibility here.

Reply:
> gson allows for some flexibility here.

Does it justify the need for new dependencies? The Jackson allow to read unquoted  field names, I think it is an easy ask to require values to be always quoted, since in many cases they have to be quoited anyway:

```
"{opensearch:"~1.2.3, =1.2.3"}
```

Reply:
@reta - Updated to use jackson instead of gson.